src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1275,7 +1280,225 @@ public void TestConflictResolverDeletedRemoteWin()
         }
 
         [Fact]
-        public void TestConflictResolverExceptionsNoneMatchDBThrown()
+        public void TestConflictResolverWrongDocID()
+        {
+            CreateReplicationConflict("doc1");
+
+            var config = CreateConfig(false, true, false);
+            config.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                var doc = new MutableDocument("wrong_id");
+                doc.SetString("wrong_id_key", "wrong_id_value");
+                return doc;
+            });
+
+            RunReplication(config, 0, 0);
+
+            using (var db = Db.GetDocument("doc1")) {
+                db.GetString("wrong_id_key").Should().Be("wrong_id_value");
+            }
+        }
```

Comment:
Do you want to verify that this doesn't return an error to the user?

Reply:
Yes, this is not an error condition since we correct it in the code base.